ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -82,20 +106,51 @@ def forward(
         sequence_length: int = 1,
     ) -> Tuple[torch.Tensor, torch.Tensor]:
         encodes = []
+        var_len_inputs = []  # The list of variable length inputs
+
         for idx, processor in enumerate(self.processors):
-            obs_input = inputs[idx]
-            processed_obs = processor(obs_input)
-            encodes.append(processed_obs)
+            if processor is not None:
```

Comment:
Can you just use an "identity" processor, instead of using None?
```
def ident(x):
  return x
```

Reply:
I can do an identity processor, but at no point will the forward pass will be used. This feels kind of strange to me.

Reply:
Sorry, I though that would simplify this, but missed that the results were getting appended to different lists.

Reply:
I tried something else. I put the EntityEmbedding in the list of processors now. I think it is cleaner, but it required me to add a method ("add_self_embedding") to EntityEmbeddings to specify the size of the self input after initialization.